spec/system/referee_interface/referee_cannot_provide_reference_if_two_already_received_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'rails_helper'
+
+RSpec.feature 'Referee can submit reference', with_audited: true do
+  include CandidateHelper
+
+  scenario 'Referee submits a reference for a candidate with relationship, safeguarding and review page' do
+    given_the_candidate_has_requested_three_references_and_i_am_the_third_referee
+    and_the_first_referee_has_responded_with_a_reference
+    and_the_second_referee_has_responded_with_a_reference
+    then_i_receive_an_email_with_a_reference_request
+    when_i_click_on_the_link_within_the_email
+    then_i_should_see_that_i_am_not_required_to_give_a_reference
+  end
+
+  def given_the_candidate_has_requested_three_references_and_i_am_the_third_referee
+    @first_reference = create(:reference, :feedback_requested, email_address: 'terri@example.com', name: 'Terri Tudor')
+    @second_reference = create(:reference, :feedback_requested, email_address: 'Bruce@Wayne.com', name: 'Bat Man')
+    @third_reference = create(:reference, :feedback_requested, email_address: 'Clark@Kent.com', name: 'Super Man')
+    @application = create(:completed_application_form,
+                          application_references: [
+                            @first_reference,
+                            @second_reference,
+                            @third_reference,
+                          ],
+                          candidate: current_candidate)
+  end
+
+  def and_the_first_referee_has_responded_with_a_reference
+    submit_reference(@first_reference)
+  end
+
+  def and_the_second_referee_has_responded_with_a_reference
+    submit_reference(@second_reference)
+  end
+
+  def submit_reference(reference)
+    # We are submitting references this way rather than via the UI because
+    # doing raises a Bullet eager loading error which is believed to be
+    # a false positive
```

Comment:
I think there are a couple of ways we could be a little clearer here:

- Setting up database state explicitly for this spec is fine, so you could remove the comment altogether. At the moment it raises more questions than it answers because the relationship between UI interaction and a Bullet error isn't obvious.
- Alternatively, we could do the UI interaction and disable bullet at the top of the spec (with an explanatory comment). This sets us up for revisiting the issue at a later date for further investigation.

Reply:
Good point. I'll go for option 1